waypoint/services/nats_service.py

```diff
@@ -106,14 +119,20 @@ async def event_generator():
                     break
 
                 try:
-                    messages = await subscription.fetch(batch=5, timeout=0.2)
+                    messages = await subscription.fetch(
+                        batch=5, timeout=0.2, heartbeat=0.02
+                    )
                     for message in messages:
                         event = orjson.loads(message.data)
+                        logger.trace("Received event: {}", event)
                         yield CloudApiWebhookEventGeneric(**event)
                         await message.ack()
-                except TimeoutError:
+                except FetchTimeoutError:
                     logger.trace("Timeout fetching messages continuing...")
                     await asyncio.sleep(0.1)
+                except TimeoutError:
+                    logger.error("FATAL: Timeout Error")
+                    raise
```

Comment:
I assume this only ends up being raised when retry logic fails too many times? Perhaps it's worth using `critical` log level, if it's appropriate. And maybe just print the exception string with the log